pkg/segment/segment.go

```diff
@@ -105,29 +108,124 @@ func (s *segmentStore) Put(ctx context.Context, path paths.Path, data io.Reader,
 	}
 
 	// puts pointer to pointerDB
-	_, err = s.pc.Put(ctx, &pr)
+	_, err = s.pdb.Put(ctx, &pr)
 	if err != nil || status.Code(err) == codes.Internal {
 		zap.L().Error("failed to put", zap.Error(err))
 		return Error.Wrap(err)
 	}
 	return nil
 }
 
-// Get retrieves a file from the erasure code client with help from overlay and pointerdb
+// Get retrieves a file using erasure code, overlay, and pointerdb clients
 func (s *segmentStore) Get(ctx context.Context, path paths.Path) (ranger.Ranger, Meta, error) {
-	m := Meta{
-		Data: []byte(""),
+	m := Meta{}
+	// TODO: remove this chunk after pointerdb client interface merged
+	gr := &ppb.GetRequest{
+		Path:   []byte(fmt.Sprintf("%s", path)),
+		APIKey: nil,
+	}
+
+	pdbRes, err := s.pdb.Get(ctx, gr)
+	if err != nil {
+		return nil, m, err
+	}
+
+	// TODO: remove this chunk after pointerdb client interface merged
+	pointer := &ppb.Pointer{}
+	err = proto.Unmarshal(pdbRes.Pointer, pointer)
+	if err != nil {
+		return nil, m, err
+	}
+
+	if pointer.Type != ppb.Pointer_REMOTE {
+		zap.L().Error("TODO: only getting remote pointers supported")
+		return nil, m, err
+	}
+
+	remoteSeg := pointer.Remote
+	var nodes []*opb.Node
+	for i := 0; i < len(remoteSeg.RemotePieces); i++ {
+		overlayRes, err := s.oc.Lookup(ctx, &opb.LookupRequest{NodeID: remoteSeg.RemotePieces[i].NodeId})
+		if err != nil {
+			return nil, m, err
+		}
+		nodes = append(nodes, overlayRes.Node)
 	}
-	return nil, m, nil
+	pid := client.PieceID(remoteSeg.PieceId)
+	ecRes, err := s.ec.Get(ctx, nodes, s.rs, pid, pointer.Size)
+	if err != nil {
+		return nil, m, err
+	}
+
+	m.Data = pointer.Metadata
+
+	return ecRes, m, nil
 }
 
-// Delete issues deletes of a file to all piece stores and deletes from pointerdb
+// Delete tells piece stores to delete a segment and deletes pointer from pointerdb
 func (s *segmentStore) Delete(ctx context.Context, path paths.Path) error {
+	// TODO: remove this chunk after pointerdb client interface merged
+	gr := &ppb.GetRequest{
+		Path:   []byte(fmt.Sprintf("%s", path)),
+		APIKey: nil,
+	}
+
+	// gets pointer from pointerdb
+	pdbRes, err := s.pdb.Get(ctx, gr)
+	if err != nil {
+		return err
+	}
+
+	// TODO: remove this chunk after pointerdb client interface merged
+	pointer := &ppb.Pointer{}
+	err = proto.Unmarshal(pdbRes.Pointer, pointer)
+	if err != nil {
+		return err
+	}
+
+	// piece store client sends delete request
+	_, err = s.ps.Delete(ctx, &pspb.PieceDelete{Id: pointer.Remote.PieceId})
```

Comment:
oh, close! which piece store is this?

we should probably add delete to the ecclient, so it knows how to farm a delete out to all 40 piece store nodes